scripts/vcf_site_level_filtering_wdl/JointVcfFiltering.wdl

```diff
@@ -1,292 +1,280 @@
+    }
+
+    output {
+        File extracted_annotations_hdf5 = ExtractVariantAnnotations.annotations_hdf5
+        File? extracted_unlabeled_annotations_hdf5 = ExtractVariantAnnotations.unlabeled_annotations_hdf5
+        File extracted_vcf = ExtractVariantAnnotations.extracted_vcf
+        File extracted_vcf_idx = ExtractVariantAnnotations.extracted_vcf_idx
+
+        Array[File] model_files = TrainVariantAnnotationsModel.model_files
+
+        Array[File] scored_vcfs = ScoreVariantAnnotations.scored_vcf
+        Array[File] scored_vcf_idxs = ScoreVariantAnnotations.scored_vcf_idx
+        Array[File?] annotations_hdf5s = ScoreVariantAnnotations.annotations_hdf5
+        Array[File?] scores_hdf5s = ScoreVariantAnnotations.scores_hdf5
+    }
 }
 
 task ExtractVariantAnnotations {
-	input {
-		String gatk_docker
-		File? gatk_override
-		File input_vcf
-		File input_vcf_index
-		String basename
-		String mode
-		String annotations
-		String resource_args
-		File? interval_list
-		Boolean use_allele_specific_annotations
-
-		Int memory_mb = 14000
-		Int command_mem = memory_mb - 1000
-	}
-	Int disk_size = ceil(size(input_vcf, "GB") + 50)
-	command {
-		set -e
-		export GATK_LOCAL_JAR=~{default="/root/gatk.jar" gatk_override}
-
-		gatk --java-options "-Xmx~{command_mem}m" \
-			ExtractVariantAnnotations \
-			-V ~{input_vcf} \
-			-O ~{basename}.~{mode} \
-			~{annotations} \
-			~{if use_allele_specific_annotations then "--use-allele-specific-annotations" else ""} \
-			~{"-L " + interval_list} \
-			--mode ~{mode} \
-			~{resource_args}
-	}
-	output {
-		File annots = "~{basename}.~{mode}.annot.hdf5"
-		File extracted_training_vcf = "~{basename}.~{mode}.vcf.gz"
-		File extracted_training_vcf_index = "~{basename}.~{mode}.vcf.gz.tbi"
-		Array[File] outputs = glob("~{basename}.~{mode}.*")
-	}
-	runtime {
-		docker: gatk_docker
-		disks: "local-disk " + disk_size + " LOCAL"
-		memory: memory_mb + " MiB"
-	}
+    input {
+        File input_vcf
+        File input_vcf_idx
+        String output_prefix
+        Array[String] annotations
+        String resource_args
+        String? extra_args
+
+        String gatk_docker
+        File? gatk_override
+
+        RuntimeAttributes runtime_attributes = {}
+    }
+
+    parameter_meta {
+        input_vcf: {localization_optional: true}
+        input_vcf_idx: {localization_optional: true}
+    }
+
+    command {
+        set -e
+        export GATK_LOCAL_JAR=~{default="/root/gatk.jar" gatk_override}
+
+        gatk --java-options "-Xmx~{default=6 runtime_attributes.command_mem_gb}G" \
+            ExtractVariantAnnotations \
+                -V ~{input_vcf} \
+                -O ~{output_prefix}.extract \
+                -A ~{sep=" -A " annotations} \
+                ~{resource_args} \
+                ~{extra_args}
+    }
+
+    runtime {
+        docker: gatk_docker
+        cpu: select_first([runtime_attributes.cpu, 1])
+        memory: select_first([runtime_attributes.machine_mem_gb, 7]) + " GB"
```

Comment:
For what it's worth the default memory requirements were too small for both the Extract and the Train tasks for a callset of ~200 samples. Not sure if that means it's worth it to increase the defaults or not for this example WDL. 

However, as we discussed in slack, I would personally vote for making the `command_mem_gb` related to `machine_mem_gb` (`machine_mem_gb = command_mem_gb +1`). It seems much simpler for the user to only have to worry about one of those arguments. If you're concerned that they should be untied due to needing extra memory on the machine outside of the java options, then maybe add a `additional_machine_mem_gb` or something similar (`machine_mem_gb = command_mem_gb + additional_machine_mem_gb`)? That way you'd still only have two memory arguments (`command_mem_gb` and `additional_machine_mem_gb`), but in most use cases the user only needs to update the `command_mem_gb`.

Reply:
Went with `command_mem_gb` and `additional_mem_gb`, thanks for the suggestion! I just left the default values as is, though---they're just meant to be an example.